bot/seasons/pride/pride_avatar.py

```diff
@@ -0,0 +1,133 @@
+    "aromantic": "aromantic",
+    "aro": "aromantic",
+    "ace": "asexual",
+    "asexual": "asexual",
+    "bigender": "bigender",
+    "bisexual": "bisexual",
+    "bi": "bisexual",
+    "demiboy": "demiboy",
+    "demigirl": "demigirl",
+    "demi": "demisexual",
+    "demisexual": "demisexual",
+    "gay": "gay",
+    "lgbt": "gay",
+    "queer": "gay",
+    "homosexual": "gay",
+    "fluid": "genderfluid",
+    "genderfluid": "genderfluid",
+    "genderqueer": "genderqueer",
+    "intersex": "intersex",
+    "lesbian": "lesbian",
+    "non-binary": "nonbinary",
+    "enby": "nonbinary",
+    "nb": "nonbinary",
+    "nonbinary": "nonbinary",
+    "omnisexual": "omnisexual",
+    "omni": "omnisexual",
+    "pansexual": "pansexual",
+    "pan": "pansexual",
+    "pangender": "pangender",
+    "poly": "polysexual",
+    "polysexual": "polysexual",
+    "polyamory": "polyamory",
+    "polyamorous": "polyamory",
+    "transgender": "transgender",
+    "trans": "transgender",
+    "trigender": "trigender"
+}
+
+
+class PrideAvatar(commands.Cog):
+    """Put an LGBT spin on your avatar!"""
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    @staticmethod
+    def crop_avatar(avatar):
+        """This crops the avatar into a circle."""
+
+        mask = Image.new("L", avatar.size, 0)
+        draw = ImageDraw.Draw(mask)
+        draw.ellipse((0, 0) + avatar.size, fill=255)
+        avatar.putalpha(mask)
+        return avatar
+
+    @staticmethod
+    def crop_ring(ring, px):
+        """This crops the ring into a circle."""
+
+        mask = Image.new("L", ring.size, 0)
+        draw = ImageDraw.Draw(mask)
+        draw.ellipse((0, 0) + ring.size, fill=255)
+        draw.ellipse((px, px, 1024-px, 1024-px), fill=0)
+        ring.putalpha(mask)
+        return ring
+
+    @commands.command(aliases=["avatarpride", "pridepfp", "prideprofile"])
+    async def prideavatar(self, ctx, option="lgbt", pixels: int = 64):
+        """
+        This surrounds an avatar with a border of a specified LGBT flag.
+
+        This defaults to the LGBT rainbow flag if none is given.
+        The amount of pixels can be given which determines the thickness of the flag border.
+        This has a maximum of 512px and defaults to a 64px border.
+        The full image is 1024x1024.
+        """
+
+        pixels = 0 if pixels < 0 else 512 if pixels > 512 else pixels
+
+        option = option.lower()
+
+        if option not in OPTIONS.keys():
+            return await ctx.send("I don't have that flag!")
+
+        flag = OPTIONS[option]
+
+        async with ctx.typing():
+
+            # Get avatar bytes
+            image_bytes = ctx.author.avatar_url.read()
+            avatar = Image.open(BytesIO(image_bytes))
+            avatar = avatar.convert("RGBA").resize((1024, 1024))
+
+            avatar = self.crop_avatar(avatar)
+
+            ring = Image.open(Path("bot", "resources", "pride", "flags", f"{flag}.png")).resize((1024, 1024))
```

Comment:
You need to ensure that the [`mode`](https://pillow.readthedocs.io/en/5.1.x/handbook/concepts.html#concept-modes) of the loaded flag matches that of the avatar, otherwise `alpha_composite()` will raise a `ValueError`.

The transgender flag, for example, loads as `PA`, and just turns into a black ring if you try `ring.convert("RGBA")`.

Reply:
I understand the requirement for the mode of the loaded flag to match, but I couldn't replicate your specific example.

For me, the flag loads as a normal `RGBA` image and works fine with the command.

I'm not sure if it's a difference between OSs or PIL versions or something.

Reply:
Surely there's a way to enforce it even if you can't replicated it on your machine. "It works for me" really doesn't help the cog not crash when deployed by the bot.

Reply:
I was able to replicate the transgender flag loading as `PA` however no matter what I tried, doing `ring.convert("RGBA")` would always work for me.

I added that line to this PR to enforce that the ring is RGBA however I cannot replicate the blank ring.

The way to enforce the mode is to convert it to the preferred mode however I cannot see why it turns into a black ring on your device, no matter how hard I've tried

Reply:
Seems like this is probably just my computer/debugger's fault. The black ring of death doesn't happen anymore.